packages/main-ui/src/services/user_service.rs

```diff
@@ -327,6 +325,41 @@ impl UserService {
     pub fn icp_address(&self) -> Option<String> {
         self.wallet().principal()
     }
+
+    pub async fn listen_for_account_changes(&mut self) {
+        let mut srv = self.clone();
+        spawn(async move {
+            if let Err(e) = KaikasWallet::listen_for_account_changes(move |new_address| {
+                spawn(async move {
+                    tracing::debug!("Account changed to: {}", new_address);
+                    srv.update_wallet_address(new_address).await;
+                });
+            })
+            .await
+            {
+                tracing::error!("Failed to listen for account changes: {:?}", e);
+            }
+        });
+    }
```

Comment:
```suggestion
    pub async fn listen_for_account_changes(&mut self) {
        let mut srv = *self;
        if let Err(e) = KaikasWallet::listen_for_account_changes(move |new_address| {
            spawn(async move {
                tracing::debug!("Account changed to: {}", new_address);
                srv.update_wallet_address(new_address).await;
            });
        })
        .await
        {
            tracing::error!("Failed to listen for account changes: {:?}", e);
        }
    }
```

why spawn wraps here?

Reply:
It was to to avoid blocking the main thread and handle ownership/lifetime but I guess it doesn't matter since we are not using `tokio spawn`